core/base/ftmTree/FTMTreeUtils.cpp

```diff
@@ -148,5 +212,286 @@ namespace ttk {
       this->getTreeBranching(branching, branchingID, nodeBranching);
     }
 
+    std::vector<idNode> FTMTree_MT::getAllRoots() {
+      std::vector<idNode> roots;
+      for(idNode node = 0; node < this->getNumberOfNodes(); ++node)
+        if(this->isRoot(node) and !this->isLeaf(node))
+          roots.push_back(node);
+      return roots;
+    }
+
+    int FTMTree_MT::getNumberOfRoot() {
+      int noRoot = 0;
+      for(idNode node = 0; node < this->getNumberOfNodes(); ++node)
+        if(this->isRoot(node) and !this->isLeaf(node))
+          ++noRoot;
+      return noRoot;
+    }
+
+    int FTMTree_MT::getNumberOfChildren(idNode nodeId) {
+      return this->getNode(nodeId)->getNumberOfDownSuperArcs();
+    }
+
+    int FTMTree_MT::getTreeDepth() {
+      int maxDepth = 0;
+      std::queue<std::tuple<idNode, int>> queue;
+      queue.push(std::make_tuple(this->getRoot(), 0));
+      while(!queue.empty()) {
+        auto tup = queue.front();
+        queue.pop();
+        idNode node = std::get<0>(tup);
+        int depth = std::get<1>(tup);
+        maxDepth = std::max(maxDepth, depth);
+        for(idNode child : this->getChildren(node))
+          queue.push(std::make_tuple(child, depth + 1));
+      }
+      return maxDepth;
+    }
+
+    int FTMTree_MT::getNodeLevel(idNode nodeId) {
+      int level = 0;
+      auto root = this->getRoot();
+      int noRoot = this->getNumberOfRoot();
+      if(noRoot != 1) {
+        std::stringstream ss;
+        ss << "problem, there is " << noRoot << " root(s)";
+        printMsg(ss.str(), debug::Priority::ERROR);
```

Comment:
you may want to simply use the `printErr` function.

Reply:
Ok! Done in the next commit.